File: ruby/matrix/matrix.rb
class Matrix
  attr_reader :rows, :columns

  def self.rotate(array)
    # Rotate 90 degrees

    width = array.first.size
    new = []

    width.times do |i|
      new << array.map { |row| row[i] }
    end

    return new
  end

  def initialize(str)
    @rows = str.split("\n").map { |row| row.split.map(&:to_i) }
    @columns = Matrix.rotate(@rows)
  end
end

File: javascript/matrix/matrix.js
export class Matrix {
    constructor(str) {
        this.rows = [];
        this.columns = [];

        // Populate rows
        for (const row of str.split("\n"))
            this.rows.push( row.split(" ").map(n => parseInt(n)) );

        // Transpose rows into columns
        for (let i = 0; i < this.rows[0].length; ++i)
            this.columns.push( this.rows.map(row => row[i]) );
    }
}
